src/data/slice.ts

```diff
@@ -231,6 +235,50 @@ const rootSlice = createSlice({
 		setActiveTabId(state, action: PayloadAction<TabKind>) {
 			state.activeTabId = action.payload;
 		},
+		changeJob(state, action: PayloadAction<'prev' | 'next'>) {
+			// this selector is expensive to calculate
+			const changeExplorerTree = selectExplorerTree(
+				state,
+				workspace.workspaceFolders?.[0]?.uri ?? null,
+			);
+
+			if (changeExplorerTree === null) {
+				return;
+			}
+
+			const index = changeExplorerTree.nodeData.findIndex((nodeDatum) => {
+				return (
+					nodeDatum.node.hashDigest ===
+					changeExplorerTree.selectedNodeHashDigest
+				);
+			});
+
+			if (index === -1) {
+				return;
+			}
+
+			const nodeData = [
+				...changeExplorerTree.nodeData.slice(index + 1),
+				...changeExplorerTree.nodeData.slice(0, index),
+			];
+
+			if (action.payload === 'prev') {
+				nodeData.reverse();
+			}
+
+			const nodeDatum =
+				nodeData.find((nodeDatum) => {
+					return nodeDatum.node.kind === 'FILE';
+				}) ?? null;
+
+			if (nodeDatum === null || nodeDatum.node.kind !== 'FILE') {
```

Comment:
No, because that `find` use does not narrow the types.

The easier way to narrow the types would be to use the following snippet:
```const node =
				nodeData
					.map(({ node }) => node)
					.find<ExplorerNode & { kind: 'FILE' }>(
						(node): node is ExplorerNode & { kind: 'FILE' } => {
							return node.kind === 'FILE';
						},
					) ?? null;

			if (node === null) {
				return;
			}

			state.changeExplorerView.focusedJobHash = node.jobHash;```

but here we'd use `.map` just for our type convenience, and it's pointless to map thru `n` elements for that purpose.